fixed regex boundary escaping

jschoch opened the pull request:
this fixes issues with emails that have special characters in them that collide with regex special characters.

Shakadak commented:
Hi @jschoch, I'm a bit confused. Could you provide examples of the issue you are mentioning ? I also don't understand why the test files where changed. Could you provide additional tests targeting your specific issue ?

jschoch commented:
if the boundary has a "?" in it it crashes

boundary="_NextPart_+5?29\786cba680f4cda8c47a92f2d7e09d"

the string is not escaped so it parses the ? as a regex special character.  


example:

```elixir
iex(1)> boundary="_NextPart_+5?29\786cba680f4cda8c47a92f2d7e09d"
"_NextPart_+5?29786cba680f4cda8c47a92f2d7e09d"

iex(2) body = "foo  --_NextPart_+5?29\786cba680f4cda8c47a92f2d7e09d  bar baz --_NextPart_+5?29\786cba680f4cda8c47a92f2d7e09d"
"foo  --_NextPart_+5?29786cba680f4cda8c47a92f2d7e09d  bar baz --_NextPart_+5?29786cba680f4cda8c47a92f2d7e09d"

#   this is not split
iex(3)> body |> String.split(~r"\s*--#{boundary}\s*")
["foo  --_NextPart_+5?29786cba680f4cda8c47a92f2d7e09d  bar baz --_NextPart_+5?29786cba680f4cda8c47a92f2d7e09d"]

iex(4)> escaped_bound = Regex.escape(boundary)
"_NextPart_\\+5\\?29786cba680f4cda8c47a92f2d7e09d"

# this is properly split
iex(5)> body |> String.split(~r"\s*--#{escaped_bound}\s*")
["foo", "bar baz", ""]

```

Shakadak commented:
Alright, that's small enough for me to accept it as is. I want to deal with the codepagex dependency update before publishing though.
